examples/howto/optimize_evoked.py

```diff
@@ -12,203 +12,278 @@
-                          weights_nmda=weights_nmda_p2,
-                          synaptic_delays=synaptic_delays_p)
-
+weights_ampa_p2 = {'L2_basket': 0.000003, 'L2_pyramidal': 1.438840,
+                   'L5_basket': 0.008958, 'L5_pyramidal': 0.1}
+net_initial.add_evoked_drive('evprox2',
+                             mu=120.,
+                             sigma=1.,
+                             numspikes=1,
+                             location='proximal',
+                             weights_ampa=weights_ampa_p2,
+                             synaptic_delays=synaptic_delays_p,
+                             n_drive_cells=n_drive_cells,
+                             cell_specific=cell_specific)
+
+# Simulate initial dipole
+n_trials = 10
+tstop = dipole_experimental.times[-1]
 with MPIBackend(n_procs=n_procs, mpi_cmd='mpiexec'):
-    init_dpl = simulate_dipole(net_init, tstop=tstop, n_trials=1)[0]
-init_dpl.scale(scale_factor)
-init_dpl.smooth(smooth_window_len)
+    dipoles_initial = simulate_dipole(
+        net_initial, tstop=tstop, n_trials=n_trials)
+
+# Smooth and scale
+window_length = 30
+scaling_factor = 3000
+for dipole in dipoles_initial:
+    dipole.smooth(window_length).scale(scaling_factor)
+
+# Plot
+fig, ax = plt.subplots(figsize=(6, 6))
+dipole_experimental.plot(ax=ax, layer='agg', show=False,
+                         color='tab:blue')
+plot_dipole(dipoles_initial.copy(), ax=ax, layer='agg', show=False,
+            color='tab:orange', average=True)
+
+# Legend
+legend_handles = [Line2D([0], [0], color='tab:blue', lw=1.0),
+                  Line2D([0], [0], color='tab:orange', lw=1.0)]
+ax.legend(legend_handles, ['experimental', 'initial'])
 
 ###############################################################################
-# Now we start the optimization!
+# 4. Define the ``set_params`` function
+# -----------------------------------
+#
+# Now we can begin the optimization!
 #
-# First, we define a function that will tell the optimization routine how to
-# modify the network drive parameters. The function will take in the Network
-# object with no attached drives, and a dictionary of the parameters we wish to
-# optimize.
+# First, we'll define a ``set_params`` function. This function tells the optimizer
+# how to apply the chosen parameters to the network during each iteration.
+#
+# Comparison of the initial dipoles reveals that the second positive peak of the
+# experimental dipole (blue) has a larger amplitude, later latency, and broader
+# temporal distribution than the corresponding peak in the initial dipole (orange).
+#
+# Let's optimize the relevant parameters for the second proximal (evprox2) drive,
+# as it is primarily responsible for this component of the dipole.
 
 
 def set_params(net, params):
 
     # Proximal 1
     net.add_evoked_drive('evprox1',
-                         mu=5.6813,
-                         sigma=20.3969,
+                         mu=26.61,
+                         sigma=2.47,
```

Comment:
If `cell_specific=False` and `n_drive_cells=1` then I think the variance only comes into play when there are multiple trials I think

If you are using 1 trial based on the earlier feedback perhaps set this to 0.0